### PR TITLE
Grigoriev Dmitrii

### PR DESCRIPTION
**Pull request by meltiser**

### CODE REVIEW

File: src/main/java/com/epam/homework/beans/JazzMusic.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public String getMusician() {
+        return musician;
+    }
+
+    @Override
+    public String getSong() {
+        return song;
+    }
+
+    @PostConstruct
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/homework/beans/JazzMusic.java]: It is highly recommended to place init method in the beginning of the class and predestroy method - in the end. It's increases readability.

File: src/main/java/com/epam/homework/task3/CustomContextListener.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.ContextClosedEvent;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.ContextStartedEvent;
+import org.springframework.context.event.ContextStoppedEvent;
+import org.springframework.context.support.AbstractApplicationContext;
+
+@Configuration
+public class CustomContextListener implements ApplicationListener {
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/homework/task3/CustomContextListener.java]: Take a look at the raw type here, we can use some generics, that makes dome diffs in the behavior

File: src/main/java/com/epam/homework/task4/CustomBeanPostProcessor.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import com.epam.homework.beans.Music;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class CustomBeanPostProcessor implements BeanPostProcessor {
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        Music beanMusic = (Music) bean;
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/homework/task4/CustomBeanPostProcessor.java]: Why you want, that all your beans will be casted to Music? Why in this method?

File: src/main/java/com/epam/homework/beans/JazzMusic.java (hunk 1 above)
Comment (meltiser) [reply src/main/java/com/epam/homework/beans/JazzMusic.java]: Fixed.

File: src/main/java/com/epam/homework/task3/CustomContextListener.java (hunk 2 above)
Comment (meltiser) [reply src/main/java/com/epam/homework/task3/CustomContextListener.java]: Added generic. I could implement ContextListener class for every event, but decided to make it all in one )

File: src/main/java/com/epam/homework/task4/CustomBeanPostProcessor.java (hunk 3 above)
Comment (meltiser) [reply src/main/java/com/epam/homework/task4/CustomBeanPostProcessor.java]: Cause my bean post proccer works only with Music instance beans. But yes, that should be fixed.
Added logic which checks if bean is an instance of Music interface. If it is, sets him a value.

**Pull request merged**